### PR TITLE
Hello PropStream and PropWriteStream

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Add PropStream and PropWriteStream for Lua API.
- PropStream has no use today, however, already ready.
- PropWriteStream can be used with `db.escapeBlob`.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -10644,7 +10694,8 @@ int LuaScriptInterface::luaPlayerSetGhostMode(lua_State* L)
 		Player* spectatorPlayer = static_cast<Player*>(spectator);
 		if (spectatorPlayer != player && !spectatorPlayer->isAccessPlayer()) {
 			if (enabled) {
-				spectatorPlayer->sendRemoveTileCreature(player, position, tile->getClientIndexOfCreature(spectatorPlayer, player));
+				spectatorPlayer->sendRemoveTileCreature(player, position,
+				                                        tile->getClientIndexOfCreature(spectatorPlayer, player));
```
Comment (ramon-bernardo) [inline src/luascript.cpp:10698]: code format...🤔

**Comment by MillhioreBT** [comment]
I think this is not the style used in Lua. For Python it would be perfect.
`obj:method_example()`
always has been: `obj:methodExample()`

**Comment by ramon-bernardo** [comment]
> I think this is not the style used in Lua. For Python it would be perfect. `obj:method_example()` always has been: `obj:methodExample()`
Thanks!
It looks weird readI8 😄

File: src/luascript.cpp [hunk 2]
```diff
@@ -18071,6 +18122,403 @@ int LuaScriptInterface::luaXmlNodeNextSibling(lua_State* L)
...
+	return 1;
+}
+
+int LuaScriptInterface::luaPropStreamReadString(lua_State* L)
+{
+	// propStream:readString(ret)
+	PropStream* propStream = getUserdata<PropStream>(L, 1);
+	if (propStream) {
+		std::string& ret = getString(L, 2);
+		lua_pushboolean(L, propStream->readString(ret));
```
Comment (ramon-bernardo) [inline src/luascript.cpp:18180]: I need a review here, mainly on PropStream.

**Comment by ranisalt** [comment]
I think this can be emulated in pure Lua with IO operations.

**Comment by ramon-bernardo** [comment]
> I think this can be emulated in pure Lua with IO operations.

You're right, I'll find in doc

**Pull request closed without merging**